evalml/automl/automl_search.py

```diff
@@ -1307,18 +1308,33 @@ def best_pipeline(self):
 
         return self._best_pipeline
 
-    def save(self, file_path, pickle_protocol=cloudpickle.DEFAULT_PROTOCOL):
+    def save(
+        self,
+        file_path,
+        pickle_type="cloudpickle",
+        pickle_protocol=cloudpickle.DEFAULT_PROTOCOL,
+    ):
         """Saves AutoML object at file path
 
         Arguments:
             file_path (str): location to save file
+            pickle_type {"pickle", "cloudpickle"}: the pickling library to use.
             pickle_protocol (int): the pickle data stream format.
 
         Returns:
             None
         """
+        if pickle_type == "cloudpickle":
+            pkl_lib = cloudpickle
+        elif pickle_type == "pickle":
+            pkl_lib = pickle
+        else:
+            raise ValueError(
+                f"`pickle_type` must be either 'pickle' or 'cloudpickle'. Received {pickle_type}"
+            )
+
         with open(file_path, "wb") as f:
-            cloudpickle.dump(self, f, protocol=pickle_protocol)
+            pkl_lib.dump(self, f, protocol=pickle_protocol)
```

Comment:
What if the user uses a cloudpickle protocol while trying to use pickle? 🤔

Reply:
Apparently cloudpickles can be opened by the regular pickling library (according to their example on their [README.md](https://github.com/cloudpipe/cloudpickle) and the doc string for `cloudpickle.py` in the attached screenshot)! I didn't know this before so that's pretty neat.

![image](https://user-images.githubusercontent.com/8752455/124139593-c3fac900-da55-11eb-8b36-cd46ed0a1417.png)

Reply:
So cool 🤩